src/Mapping/Operator/Simple/LoadDataObject.php

```diff
@@ -74,17 +80,28 @@ public function process($inputData, bool $dryRun = false)
                 $object = DataObject::getById(trim($data));
             } elseif ($this->loadStrategy === self::LOAD_STRATEGY_ATTRIBUTE) {
                 if ($this->attributeName) {
-                    $getter = 'getBy' . $this->attributeName;
-                    $class = ClassDefinition::getById($this->attributeDataObjectClassId);
-                    if (empty($class)) {
-                        throw new InvalidConfigurationException("Class `{$this->attributeDataObjectClassId}` not found.");
-                    }
-                    $className = '\\Pimcore\\Model\\DataObject\\' . ucfirst($class->getName());
-
-                    if ($this->attributeLanguage) {
-                        $object = $className::$getter($data, $this->attributeLanguage, 1);
+                    if ($this->partialMatch) {
+                        $listClassName = $className . '\\Listing';
+                        $listing = new $listClassName();
+                        $listing->setCondition($this->attributeName . ' like ' . $listing->quote("%".$data."%"));
+                        foreach ($listing as $listObject) {
```

Comment:
why iterating through all of the elements? 
why not do `$listing->setLimit(1)` and just return the first found element?

Reply:
True, it will still only pick one matched object. I changed the code according to your suggestion.
P.S. is there a way to get just one object from the Listing without using foreach? I tried to search for an "array_pop" kind of an alternative, but didn't find one. But I guess, in any case foreach still does the trick :)

Reply:
it should be possible to use 
```php
$elements = $listing->load();
$object = reset($elements);
```

or 

```
$object = $listing->load()[0] ?? null;
```